packages/relay/src/lib/services/debugService/index.ts

```diff
@@ -0,0 +1,261 @@
+import { EthImpl } from '../../eth';
+const SUCCESS = 'SUCCESS';
+
+/**
+ * Represents a DebugService for tracing and debugging transactions and debugging
+ *
+ * @class
+ * @implements {IDebugService}
+ */
+export class DebugService implements IDebugService {
+  /**
+   * The interface through which we interact with the mirror node
+   * @private
+   */
+  private readonly mirrorNodeClient: MirrorNodeClient;
+
+  /**
+   * The logger used for logging all output from this class.
+   * @private
+   */
+  private readonly logger: Logger;
+
+  /**
+   * The commonService containing useful functions
+   * @private
+   */
+  private readonly common: CommonService;
+
+  public readonly debugTraceTransaction = 'debug_traceTransaction';
+
+  /**
+   * Creates an instance of DebugService.
+   *
+   * @constructor
+   * @param {MirrorNodeClient} mirrorNodeClient - The client for interacting with the mirror node.
+   * @param {Logger} logger - The logger used for logging output from this class.
+   * @param {CommonService} common - The common service containing useful functions.
+   */
+  constructor(mirrorNodeClient: MirrorNodeClient, logger: Logger, common: CommonService) {
+    this.logger = logger;
+    this.common = common;
+    this.mirrorNodeClient = mirrorNodeClient;
+  }
+
+  /**
+   * Checks if the Debug API is enabled
+   * @public
+   */
+  public static readonly isDebugAPIEnabled = process.env.DEBUG_API_ENABLED;
+
+  static requireDebugAPIEnabled(): void {
+    if (!process.env.DEBUG_API_ENABLED || process.env.DEBUG_API_ENABLED !== 'true') {
+      throw predefined.UNSUPPORTED_METHOD;
+    }
+  }
+
+  /**
+   * Trace a transaction for debugging purposes.
+   *
+   * @async
+   * @param {string} transactionHash - The hash of the transaction to be traced.
+   * @param {TracerType} tracer - The type of tracer to use (either 'CallTracer' or 'OpcodeLogger').
+   * @param {object} tracerConfig - The configuration object for the tracer.
+   * @param {string} [requestIdPrefix] - An optional request id.
+   * @throws {Error} Throws an error if the specified tracer type is not supported or if an exception occurs during the trace.
+   * @returns {Promise<any>} A Promise that resolves to the result of the trace operation.
+   *
+   * @example
+   * const result = await debug_traceTransaction('0x123abc', TracerType.CallTracer, {"tracerConfig": {"onlyTopCall": false}}, some request id);
+   */
+  async debug_traceTransaction(
+    transactionHash: string,
+    tracer: TracerType,
+    tracerConfig: object,
+    requestIdPrefix?: string,
+  ): Promise<any> {
+    this.logger.trace(`${requestIdPrefix} debug_traceTransaction(${transactionHash})`);
+    try {
+      DebugService.requireDebugAPIEnabled();
+      if (tracer === TracerType.CallTracer) {
+        return await this.callTracer(transactionHash, tracerConfig, requestIdPrefix);
+      } else if (tracer === TracerType.OpcodeLogger) {
+        throw predefined.UNSUPPORTED_METHOD;
```

Comment:
nit: please add a message to make it clear it's not supported for callTracer types

Reply:
@Nana-EC You mean its not supported for opcode logger or?

Reply:
AH yes I meant opcode.
Basically as coded it'll be a general error message. 
I'm suggesting it might be good to add a Text highlighting it'd just this type (opcode) of tracing that isn't supported.